packages/loader/container-loader/src/loader.ts

```diff
@@ -452,6 +452,20 @@ export class Loader implements IHostLoader {
 			request.headers[LoaderHeader.cache] !== false &&
 			pendingLocalState === undefined;
 		const fromSequenceNumber = request.headers[LoaderHeader.sequenceNumber] ?? -1;
+		const opsBeforeReturn = request.headers[LoaderHeader.loadMode]?.opsBeforeReturn as
+			| string
+			| undefined;
+		const shouldFreeze = request.headers[LoaderHeader.loadMode]?.freezeAfterLoad === true;
+
+		if (opsBeforeReturn === "sequenceNumber" && fromSequenceNumber < 0) {
+			// If opsBeforeReturn is set to "sequenceNumber", then fromSequenceNumber should be set to a non-negative integer.
+			// If it is negative it was either left undefined or set to a negative value. Either way we should throw an error.
+			throw new Error("sequenceNumber must be set to a non-negative integer");
+		} else if (opsBeforeReturn !== "sequenceNumber" && fromSequenceNumber !== -1) {
+			// If opsBeforeReturn is not set to "sequenceNumber", then fromSequenceNumber should be set to -1 (default value).
+			// In this case, we should throw an error since opsBeforeReturn is not explicitly set to "sequenceNumber".
+			throw new Error('opsBeforeReturn must be set to "sequenceNumber"');
+		}
```

Comment:
Looking at the logic above, I want to say that there should be no fromSequenceNumber argument. Instead opsBeforeReturn  should be current type | number.

Reply:
This was an idea I experimented with as well. Ultimately I liked the idea of using `LoaderHeader.sequenceNumber` since it already exists and can conflict with `opsBeforeReturn`. By adding a number option to `opsBeforeReturn` the two load options have even more collision than before.